colcon_core/package_descriptor.py

```diff
@@ -100,24 +101,24 @@ def get_recursive_dependencies(
         dependencies = set()
         while queue:
             # pick one dependency from the queue
-            name = queue.pop()
+            dep = queue.pop()
```

Comment:
Why should this variable be renamed?

Reply:
I renamed it because `name` makes it sound like a string, which it is not. The wording in `DependencyDescriptor` makes it sound like the long-term plan is to break the `str` inheritance, which would leave this naming not just misleading, but wrong.

Reply:
The reason I did so in this PR rather than another, is that this change causes that value to always be a `DependencyDescriptor` instance, where before it was _usually_ a string.

Reply:
:+1: